File: Javascript/Algorithms/876.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var middleNode = function(head) {
    let mid = head,
        inc = false;
    while (head = head.next) {
        if (inc = !inc){
            mid = mid.next;
        }
    }
    return mid;
};

File: C/Algorithms/876.c
// Definition for singly-linked list.
struct ListNode {
  int val;
  struct ListNode *next;
};

struct ListNode *middleNode(struct ListNode *head) {
  struct ListNode *mid = head;
  int inc = 0;
  while ((head = (*head).next)) {
    if ((inc = !inc))
      mid = (*mid).next;
  }
  return mid;
}
